src/carma-platform/guidance_scripts.py

```diff
@@ -163,5 +179,461 @@ def run_crosstrack_analysis(
 
     return (is_passed, stats, plt.gcf(), cross_tracks, timestamps)
 
+def process_actual_path(odom_data):
+    """Helper function that processes actual path data."""
+    return np.array([[point[0], point[1]] for point in odom_data])
+
+def process_planned_path(traj_plans):
+    """Helper function that processes planned path data with duplicate removal."""
+    planned_path = []
+    last_planned_point = None
+
+    for plan in traj_plans:
+        for point in plan:
+            if last_planned_point is None:
+                planned_path.append(point)
+                last_planned_point = point
+            else:
+                dist = np.linalg.norm(np.array(point) - np.array(last_planned_point))
+                if dist > 0.25:  # 0.25m threshold
+                    planned_path.append(point)
+                    last_planned_point = point
+                    break
+
+    return np.array(planned_path)
+
+def run_turn_accuracy_analysis(
+    mcap_path,
+    error_threshold_to_pass_meter=2.0,
+    start_time=None,
+    end_time=None,
+    save_stats_dir=None,
+    save_data_dir=None,
+    save_plot_dir=None,
+):
+    """
+    Analyzes turn accuracy by comparing actual path to planned trajectory using spline fitting.
+
+    Args:
+        mcap_path: Path to MCAP file
+        start_time: Time to start the analysis
+        end_time: Time to end the analysis
+        save_stats_dir: Directory to save analysis stats
+        save_data_dir: Directory to save extracted data
+        save_plot_dir: Directory to save generated plots
+    Deps:
+        Topics: [/localization/current_pose, /localization/current_pose]
+        Msgs: carma_planning_msgs
+    """
+    # Extract actual and planned paths
+    actual_path = []
+    planned_path = []
+
+    # Extract messages from MCAP
+    topics = ["/localization/current_pose", "/guidance/plan_trajectory"]
+    extracted_data = extract_mcap_data(
+        mcap_path,
+        topics,
+        start_time=start_time,
+        end_time=end_time,
+        field_extractors={
+            "/localization/current_pose": lambda msg: (
+                msg.pose.position.x,
+                msg.pose.position.y,
+            ),
+            "/guidance/plan_trajectory": lambda msg: [
+                (p.x, p.y) for p in msg.trajectory_points[1:]
+            ],  # Skip first point
+        },
+    )
+
+    # Process actual path
+    timestamps, odom = extracted_data[topics[0]]
+
+    actual_path = process_actual_path(odom)
+
+    # Process planned path with duplicate removal
+    # timestamp source here is not too important
+    _, traj_plans = extracted_data[topics[1]]
+
+    planned_path = process_planned_path(traj_plans)
+
+    # Fit spline to planned path
+    t = np.linspace(0, 1, len(planned_path))
+    cs_x = CubicSpline(t, planned_path[:, 0])
```

Comment:
This is so that we get slightly more accurate errors distances. We sampled points from the trajectory at least 0.25 meter away from each other, but we don't know what the actual distances turn out to be. So I resampled from the spline 5 times more points to compare distances to odom for more accurate distances.